runtime/tools/spec-builder/src/genesis-builder.js

```diff
@@ -14,6 +14,11 @@ class GenesisBuilder {
         if (!(genesisConfig && genesisConfig.genesis && genesisConfig.genesis.runtime && genesisConfig.genesis.runtime.system)) {
             throw new Error('Invalid Genesis config provided');
         }
+        if (genesisConfig.genesis.runtime.balances){
```

Comment:
I think that we can move the check for `balances.length` into the `balances` class. In there we have a check:
```JavaScript
if (!balancesArray) {
    throw new Error("Balances: No balances array provided")
}
```

Having the check there will separate the concerns of the `GenesisBuilder` and the `Balances` class. Everyone should be responsible for its own things so changing the if in the balances to:
```JavaScript
if (!Array.isArray(balancesArray) {
    throw new Error("Balances: No balances array provided")
}
```

I do not even think that we need to check for the array. If someone wants to provide an empty array, why would that be invalid?

Reply:
Sorry, I might have not described it properly. -> In my opinion we steel need to check that the `balancesArray` is actually an array, but we don't have to check whether it has any elements in it. That is why I suggested you to check for `Array.isArray` instead of `!balancesArray`.
That way the invalid input of the nested balances will be caught and the only valid input would be the following:
```JavaScript
balances: {
    balances: [ ] //<- Should definitely be array and not f.e object, string or smth else
}
```